docs/src/content/en/reference/tools/mcp-server.mdx

```diff
@@ -5,48 +5,143 @@ description: API Reference for MCPServer - A class for exposing Mastra tools and
+    },
+    {
+      name: "id",
+      type: "string",
+      isOptional: true,
+      description: "Optional unique identifier for the server. If not provided, a UUID will be generated. This ID is considered final and cannot be changed by Mastra if provided.",
+    },
+    {
+      name: "description",
+      type: "string",
+      isOptional: true,
+      description: "Optional description of what the MCP server does.",
+    },
+    {
+      name: "repository",
+      type: "Repository", // { url: string; source: string; id: string; }
+      isOptional: true,
+      description: "Optional repository information for the server's source code.",
+    },
+    {
+      name: "releaseDate",
+      type: "string", // ISO 8601
+      isOptional: true,
+      description: "Optional release date of this server version (ISO 8601 string). Defaults to the time of instantiation if not provided.",
+    },
+    {
+      name: "isLatest",
+      type: "boolean",
+      isOptional: true,
+      description: "Optional flag indicating if this is the latest version. Defaults to true if not provided.",
+    },
+    {
+      name: "packageCanonical",
+      type: "'npm' | 'docker' | 'pypi' | 'crates' | string",
+      isOptional: true,
+      description: "Optional canonical packaging format if the server is distributed as a package (e.g., 'npm', 'docker').",
+    },
+    {
+      name: "packages",
+      type: "PackageInfo[]",
+      isOptional: true,
+      description: "Optional list of installable packages for this server.",
+    },
+    {
+      name: "remotes",
+      type: "RemoteInfo[]",
+      isOptional: true,
+      description: "Optional list of remote access points for this server.",
     },
   ]}
 />
 
-For example, here's how you might create a new `MCPServer` instance:
+## Exposing Agents as Tools
 
-```typescript
-import { MCPServer } from "@mastra/mcp";
-import { weatherTool } from "./tools"; // Assuming you have a weather tool defined in this file
+A powerful feature of `MCPServer` is its ability to automatically expose your Mastra Agents as callable tools. When you provide agents in the `agents` property of the configuration:
 
-const server = new MCPServer({
-  name: "My Weather Server",
-  version: "1.0.0",
-  tools: { weatherTool },
-});
-```
+- **Tool Naming**: Each agent is converted into a tool named `ask_<agentKey>`, where `<agentKey>` is the key you used for that agent in the `agents` object. For instance, if you configure `agents: { myAgentKey: myAgentInstance }`, a tool named `ask_myAgentKey` will be created.
+
+- **Tool Functionality**:
+    - **Description**: The generated tool's description will be in the format: "Ask agent '<AgentName>' a question. Original agent instructions: <first 500 chars of agent's instructions, or 'Not available'>".
```

Comment:
I think we need a `new Agent({ description: "" })` for some other things (A2A), worth adding in this PR and using it instead of instructions

Reply:
If an agent is added to MCPServer and it doesn't have a description we can throw an error saying how to add one + a docs link